lib/puppet/transaction/report.rb

```diff
@@ -469,6 +469,12 @@ def calculate_time_metrics
 
     metrics[TOTAL] = metrics.values.inject(0) { |a,b| a+b }
 
+    # If we report any time metrics that have such a minuscule value that json
+    # can't parse it, we get failures when we try to submit these metrics to
+    # puppetdb. This ensures that we don't have any values in scientific
+    # notation.
+    metrics.values.map! { |val| "%0.5f" % val if val.is_a?(Float) }
+
```

Comment:
I think we want to fix this in the `Puppet::Util::Metrics#to_data_hash` method so we always serialize float values to some max precision (otherwise we'll have the same problem if we were to use `Puppet::Util::Metrics` somewhere else.